### PR TITLE
Improve Plugin documentation

### PR DESCRIPTION
**Pull request by JorjMcKie**
# Better doxygen suport using doxypypy

Thank your for contributing to Nuitka!

!! Please check that you select the **develop branch** (details see below link) !!

Before submitting a PR, please review the guidelines:
https://github.com/kayhayen/Nuitka/blob/master/CONTRIBUTING.md

### What does this PR do?
Changed the documentation format back to standard Python ``__doc__`` style. This affects the 3 previously documented plugin modules `TkinterPlugin.py`, `NumpyPlugin.py` and `PluginBase.py`.
There are no active code changes in these modules.

### Why was it initiated? Any relevant Issues?
After locating package `doxypypy`, which is able to transform Python ``__doc__`` into `doxygen` format, we are switching to this solution for a number of obvious good reasons.
This entailed a minor change to `doc/Doxyfile` which now will use a filter to preprocess Python sources by invoking `doxypypy`.
**Before creating the next doxygen** API doc, this tool, `doxypypy`, **must be installed** (standard PIP install). The repo's root directory has two scripts `py-file.bat` (Windows) and `py-file.sh` (Linux) for invoking `doxypypy`. `py-file` must be locatable by `doxygen`. When executing ``doxygen doc/Doxyfile`` from the repo's root, everything should be good.
The Linux version has **not yet been tested**!

### PR Checklist
- [x] Correct base branch selected? `develop` for new features and bug fixes too. If it's
      part of a hotfix, it will be moved to ``master`` during it.
- [x] All tests still pass. Check the developer manual about ``Running the Tests``. There
      are Travis tests that cover the most important things however, and you are
      welcome to rely on those, but they might not cover enough.
- [ ] Ideally new features or fixed regressions ought to be covered via new tests.
- [x] Ideally new or changed features have documentation updates.

### CODE REVIEW

File: UserPlugin-Creation.rst [hunk 1]
```diff
@@ -39,12 +39,12 @@ named "trace" is active::
...
 
-    class Usr_Plugin(UserPluginBase):
+    class MyPlugin(UserPluginBase):
 
-        plugin_name = __file__  # or __module__
+        plugin_name = __file__  # or __module__ or __name__
 
         def __init__(self):
-            # demo only: extract and display its list of options
+            # demo only: extract and display options list
```
Comment (bksahu) [inline UserPlugin-Creation.rst]: Hi. I think this comment should be "`extract and display the provided option with the plugin`" or something like that. Ref. https://github.com/Nuitka/Nuitka/blob/a9443b8eeb661ab70fa2ddb9c29c85192c78e391/nuitka/Options.py#L421

File: nuitka/plugins/standard/NumpyPlugin.py [hunk 2]
```diff
@@ -119,7 +133,7 @@ def get_numpy_core_binaries():
     # Anything found here will have to land in the dist folder, because there
     # just is no logical other place, and hope for the best ...
     # TODO: not supported yet!
-    if is_win:
+    if isWin32Windows():
```
Comment (kayhayen) [inline nuitka/plugins/standard/NumpyPlugin.py]: Good catch!

**Comment by JorjMcKie** [comment]
@kayhayen, I am out of recipes now.
WTF should I do to get over this message `nuitka/plugins/PluginBase.py:393 R0201 no-self-use NuitkaPluginBase.onModuleEncounter Method could be a function`?

**Comment by kayhayen** [comment]
@JorjMcKie seems you already addressed that `no-self-use`. 

Just to state briefly what's on my mind about this, it is that PyLint wants you to avoid methods (i.e. `self` argument having functions in a class) when they don't use it, but rather make you use `staticmethod` for that.

I have put plenty of these in Nuitka: '# Virtual method, pylint: disable=no-self-use,unused-argument' which does the job in case it's an abstract base class.

Otherwise, PyLint is really hard to conquer sometimes.

Also, remember you can fight locally if you execute `bin/check-nuitka-with-pylint` or `bin/check-nuitka-with-pylint3`, but those are using `#!` to find the Python to use, so no difference on Windows there.

**Comment by JorjMcKie** [comment]
Thanks a lot! Just made this change and hope for the best.
And I did use those other cleaners, as I have now understood the purpose of all those guys … but as you say: Windows differences.

Anyway, I’ll get there sooner or later. It’s a new day.

Jorj

________________________________
Von: Kay Hayen <notifications@github.com>
Gesendet: Saturday, February 23, 2019 5:27:18 AM
An: Nuitka/Nuitka
Cc: Jorj X. McKie; Mention
Betreff: Re: [Nuitka/Nuitka] Improve Plugin documentation (#275)


@JorjMcKie<https://github.com/JorjMcKie> seems you already addressed that no-self-use.

Just to state briefly what's on my mind about this, it is that PyLint wants you to avoid methods (i.e. self argument having functions in a class) when they don't use it, but rather make you use staticmethod for that.

I have put plenty of these in Nuitka: '# Virtual method, pylint: disable=no-self-use,unused-argument' which does the job in case it's an abstract base class.

Otherwise, PyLint is really hard to conquer sometimes.

Also, remember you can fight locally if you execute bin/check-nuitka-with-pylint or bin/check-nuitka-with-pylint3, but those are using #! to find the Python to use, so no difference on Windows there.

**Comment by kayhayen** [comment]
I am already very excited to integrate this @JorjMcKie  :-)

**Comment by JorjMcKie** [comment]
Have any opinion about this new exit support?
I.e. your guidance to onStandaloneDistributionFinished, etc.?
Wait for the next release?


Jorj

________________________________
Von: Kay Hayen <notifications@github.com>
Gesendet: Saturday, February 23, 2019 6:26:33 AM
An: Nuitka/Nuitka
Cc: Jorj X. McKie; Mention
Betreff: Re: [Nuitka/Nuitka] Improve Plugin documentation (#275)


I am already very excited to integrate this @JorjMcKie<https://github.com/JorjMcKie> :-)

**Comment by JorjMcKie** [comment]
hooorray!

**Comment by kayhayen** [comment]
Congratulations. 

I think we should work through your plugins one by one, making sure they work perfectly well and do their expected thing just nicely. The improved plugin documentation is going to allow a lot of people to get their feets wet, and I will use your incremental improvements, to generate news later on too.

Now I need to check out, how to integrate this into my Buildbot, which is bound to happen over the weekend. I am plowing through GSoC 2019 student applications with priority though, so don't shoot me, even if I delay this well deserving action for a bit longer in the end.

**Pull request merged**